src/main/java/org/b612foundation/adam/opm/OrbitParameterMessage.java

```diff
@@ -103,12 +105,21 @@ public OrbitParameterMessage setSpacecraft(SpacecraftParameters spacecraft) {
     return this;
   }
 
-  public CovarianceMatrix getCovariance() {
-    return covariance;
+  public CartesianCovariance getCartesianCovariance() {
+    return cartesianCovariance;
   }
 
```

Comment:
Would it make sense to have some kind of error checking so that we would throw an error on:

- if Cartesian state vector and Keplerian covariance are both set
- if Keplerian elements and Cartesian covariance are both set
- if both Cartesian covariance and Keplerian covariance are set

Reply:
By OPM spec Cartesian always has to be set. We are sort of violating that in how we are using it in the code in that we don't require users to do that. Because the spec allows for both to be set I was thinking we should allow them both to be set as well. The bigger question is if we don't have Keplerian should we allow them to pass around Keplerian Covariance. I'm not sure.

Reply:
Hmm, ok. For now, could you document? I know that we will have validation in our API, but since this is in the astrodynamics repo (and could be used separately from the ADAM API), it might be helpful to note the behavior if different combinations of the Cartesian/Keplerian and covariance are provided.